Automattic/woocommerce-payments
# Fixed issue with subscription payment switch by adding flag to UPE pa…

Fixes #3771 

#### Changes proposed in this Pull Request



I did a lot of testing, tried different scenarios, and proposed changes in code that @RadoslavGeorgiev suggested, but at the end of the day, the simple fix that is included in this PR solves all issues. I performed various tests and even recorded a whole process using the latest woocommerce/wordpress version with two different block-based themes, and everything worked well on my end.






#### Testing instructions

1. Make sure that you have subscription with added payment method
2. Go to page where you manage subscription and change payment method
3. Enter new credit card (you can use Stripe testing cards) into form where you're supposed to add a new card
4. After submitting form, double check is new credit card added to the existing subscription

*

-------------------



**Post merge**
